### PR TITLE
API v2 Part III

### PR DESCRIPTION
**Pull request by rowanseymour**
* Moves api/v1 tests to their own module
* Ensures we're using new URLs for channels where possible
* Simplifies sitemap configuration
* Adds stub for api/v2

Requires https://github.com/nyaruka/textit/pull/17

### CODE REVIEW

File: temba/api/tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
+# -*- coding: utf-8 -*-
```
Comment (rowanseymour) [inline temba/api/tests/test_models.py]: *api/tests.py* is now split into *api/tests/test_models.py* and *api/tests/test_v1.py* to make room for API v2 tests in *api/tests/test_v2.py*

File: temba/api/authentication.py [hunk 2]
```diff
@@ -26,9 +26,6 @@ def authenticate_credentials(self, key):
             # set the org on this user
             token.user.set_org(token.org)
 
-            return (token.user, token)
+            return token.user, token
 
         raise exceptions.AuthenticationFailed('User inactive or deleted')
-
-    def authenticate_header(self, request):
```
Comment (rowanseymour) [inline temba/api/authentication.py]: This is the same as the function in the super class so not needed

File: temba/api/v1/urls.py [hunk 3]
```diff
@@ -12,29 +12,29 @@
 
 urlpatterns = [
     url(r'^$', api, name='api'),
```
Comment (rowanseymour) [inline temba/api/v1/urls.py]: This stays as 'api' and we switch it to point to the v2 docs one day in the distant future.

Comment (norkans7) [inline temba/api/v1/urls.py]: this is currently /api/v1 my be its name can be api.v1 and add a reverse name of api that map to /api/ and which will always be the lastest version

File: templates/api/api_explorer.haml [hunk 4]
```diff
@@ -16,7 +16,7 @@
         %span.divider &raquo;
 
       %li
-        %a.active{href:"{% url 'api.explorer' %}"} Explorer
+        %a.active{href:"{% url 'api.v1.explorer' %}"} Explorer
```
Comment (norkans7) [inline templates/api/api_explorer.haml]: wondering if this will not force us to duplicate the templates for every version?

Comment (rowanseymour) [inline templates/api/api_explorer.haml]: There's only template (this one) tied to the versioned API

File: templates/rest_framework/api.html [hunk 5]
```diff
@@ -64,7 +64,7 @@
     {% endblock %}
 
     <!-- Content -->
-    <a class="btn btn-primary pull-right" href='{% url "api.explorer" %}' style="margin-left: 10px">API Explorer</a>
+    <a class="btn btn-primary pull-right" href='{% url "api.v1.explorer" %}' style="margin-left: 10px">API Explorer</a>
```
Comment (norkans7) [inline templates/rest_framework/api.html]: I think this explorer url needs to be a variable in the context

File: templates/api/api_explorer.haml (hunk 4 above)
Comment (norkans7) [inline templates/api/api_explorer.haml]: I think this url can be a variable in the context

File: temba/api/v1/urls.py (hunk 3 above)
Comment (rowanseymour) [inline temba/api/v1/urls.py]: What do you mean by add a reverse name? Seems simpler to just keep this pointing to the current api.

Comment (norkans7) [inline temba/api/v1/urls.py]: I mean textit.in/api/ be always the latest (currently gives a 404) then have both textit.in/api/v1 an textit.in/api/v2 to their respective versions as long as the former is not disabled

Comment (rowanseymour) [inline temba/api/v1/urls.py]: Would be nice if that wasn't a 404. I think it should be redirect to the current api version docs.

File: templates/rest_framework/api.html (hunk 5 above)
Comment (rowanseymour) [inline templates/rest_framework/api.html]: This template is now v1 specific

**Comment by nicpottier** [comment]
For consistency should we change the reverse URLs in channels/urls.py which are pointing to v1 endpoints? (I think they are named ```api.verboice_handler``` vs ```api.v1.verboice_handler```)

Looks good otherwise though, let's get it out.

**Comment by rowanseymour** [comment]
I think if anything we should get rid of those URL names, no? No-one should be using them

**Comment by nicpottier** [comment]
Ya, good point, I'd be down for just removing those URL names.

**Comment by nicpottier** [comment]
Yarp.

**Pull request merged**